src/Stratis.Bitcoin.Features.Consensus.Tests/TimeSyncBehaviorTest.cs

```diff
@@ -28,173 +28,283 @@ public class TimeSyncBehaviorTest
-
-                // These samples will change adjusted time.
-                new object[] { false, true,  1280, TimeSpan.FromSeconds(-1),        IPAddress.Parse("2001:0db8:85a3:0000:0000:8a2e:0370:7334"), },
-                new object[] { true,  true,  3560, TimeSpan.FromSeconds(23.6),      IPAddress.Parse("::1"),                                     },
-                new object[] { true,  true,  3560, TimeSpan.FromSeconds(236),       IPAddress.Parse("12.2.3.5"),                                },
-                new object[] { false, true,  1236, TimeSpan.FromSeconds(1.236),     IPAddress.Parse("13.2.3.5"),                                },
-                new object[] { true,  true,  1236, TimeSpan.FromSeconds(-1000),     IPAddress.Parse("14.2.3.5"),                                },
-                new object[] { true,  true,  1236, TimeSpan.FromSeconds(-4.9236),   IPAddress.Parse("15.2.3.5"),                                },
-                new object[] { false, true,   118, TimeSpan.FromSeconds(-4444.444), IPAddress.Parse("16.2.3.5"),                                },
+                TestSample.Outbound(true,  false, false,    0, TimeSpan.FromSeconds(3.56),      new IPAddress(1)),
+                TestSample.Inbound(true,   false, false,    0, TimeSpan.FromSeconds(13.123),    new IPAddress(1)),
+                TestSample.Outbound(false, false, false,    0, TimeSpan.FromSeconds(7.123),     new IPAddress(1)), // ip address already used for outbound.
+                TestSample.Outbound(true,  false, false,    0, TimeSpan.FromSeconds(26.0),      IPAddress.Parse("2000:0db8:85a3:1232:0000:8a2e:0370:7334")),
+                TestSample.Outbound(false, false, false,    0, TimeSpan.FromSeconds(260),       IPAddress.Parse("2000:0db8:85a3:1232:0000:8a2e:0370:7334")), // ip address already used for outbound.
+                TestSample.Outbound(true,  false, false,    0, TimeSpan.FromSeconds(-2126.0),   new IPAddress(2)),                           
+                TestSample.Inbound( true,  false, false,    0, TimeSpan.FromSeconds(-391),      new IPAddress(3)),                           
+
+                // These samples will change adjusted time because next outbound is the 4th outbound and we are under the limits.
+                TestSample.Outbound(true,  false, false, 1280, TimeSpan.FromSeconds(-1),        new IPAddress(4)),  // 2 inbound, 4 outbound. 2/4 * 3 = 1.5 -> ceil -> 2 of each outbound   { -2126000, -2126000, -391000, -1000, -1000, 3560, 3560, 13123, 26000, 26000 } -> median is 1280ms.
+                TestSample.Inbound( true,  false, false, 3560, TimeSpan.FromSeconds(23.6),      new IPAddress(5)),  // 3 inbound, 4 outbound. 3/4 * 3 = 2.25 -> ceil -> 3 of each outbound  { -2126000, -2126000, -2126000, -391000, -1000, -1000, -1000, 3560, 3560, 3560, 13123, 23600, 26000, 26000, 26000 } -> median is 3560ms.                     
```

Comment:
You are inconsistent with the previous group about your spacing

Reply:
the leading space is a typo yes